### DISCUSSION TITLE
Step 1 roadmap discussion

### DISCUSSION THREAD

**Discussion by mikiya1991** [Umbrella-M/gpgpu#1] [General]
第一阶段基本作为任务规划，知识储备，分为几个方面：

common：
1. chisel语言研究使用
2. 编译测试平台搭建

sw：
1. ISA设计调研，是否使用Risc-v的gpu拓展 （在此之上，以后才会到编译器和triton）

hw：
1. 调度器设计
4. 运算单元设计
5. 存储系统的设计

需要每个开发，找个子项进行负责调研。选哪个项目，看个人的兴趣。
然后大家的研究要有文档或者代码的输出。

因为前期任务不急。我们暂定2周时间为一个阶段如何？

1. platform setup: https://github.com/Umbrella-M/gpgpu/issues/3
2. risc-v:  https://github.com/Umbrella-M/gpgpu/issues/2

**Comment by mikiya1991** [comment]
我先做chisel和平台搭建的调研。

结果输出为：
- 基本的仓库编译/test会可以起来。
- 给一些chisel语言的example在doc文件夹

**Reply by hardway2008** [reply]
https://github.com/THU-DSP-LAB/ventus-gpgpu  可以参考这个项目么？

**Reply by mikiya1991** [reply]
可以啊，要不你研究下那个项目的？看看哪些可以借鉴？

**Reply by mikiya1991** [reply]
我参照它来吧

**Reply by hardway2008** [reply]
好的
